### PR TITLE
dulwich: 0.10.1a -> 0.12.0

### PR DESCRIPTION
**Pull request by k0ral**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -9126,16 +9126,19 @@ in modules // {
...
-      sha256 = "02rknqarwy7p50693cqswbibqwgxzrfzdq4yhwqxbdmhbsmh0rk6";
+      sha256 = "1ihc1bdgxj7i068mhhmkzar56r2vdcj68w0dnsm7aqgcgvrp144g";
     };
 
     # Only test dependencies
     buildInputs = with self; [ pkgs.git gevent geventhttpclient mock fastimport ];
 
+    # UnicodeEncodeError: 'ascii' codec can't encode character u'\xf6' in position 13: ordinal not in range(128)
+    doCheck = false;
+
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: can be fixed by setting LC_ALL and adding pkgs.glibcLocales.
For examples check other Python packages.

Comment (k0ral) [inline pkgs/top-level/python-packages.nix]: Fixed in next commit, thanks.

**Comment by FRidh** [comment]
can you squash them

**Comment by k0ral** [comment]
Done.

**Pull request merged**